bindings/go/evmc/evmc.go

```diff
@@ -288,3 +288,22 @@ func bytesPtr(bytes []byte) *C.uint8_t {
 	}
 	return (*C.uint8_t)(unsafe.Pointer(&bytes[0]))
 }
+
+type TxInitcode struct {
+    hash Hash
+    code []byte
+}
+
+func evmcTxInitcodes(initcodes []TxInitcode) *C.evmc_tx_initcode {
+    if len(initcodes) == 0 {
+        return nil
+    }
+
+	out := make([]C.evmc_tx_initcode, len(initcodes))
+	for i := 0; i < len(initcodes); i++ {
+		out[i].hash = evmcBytes32(initcodes[i].hash)
+		out[i].code = bytesPtr(initcodes[i].code)
+		out[i].code_size = C.size_t(len(initcodes[i].code))
+	}
+	return &out[0]
```

Comment:
Are you sure you can pass this GC pointer to C?

And the formatting is wrong.

Reply:
I think it make sense to add a Go unit test where we actually pass some initcodes. Go has a runtime inspection of passed pointers and will likely report errors if we are doing something wrong.

Reply:
I think it's definitely wrong, but digging deeper into this I am now confused, because when returning output data from Go host `call` to C side it does something similar:
https://github.com/ethereum/evmc/blob/954944a4286a9e3758c80c05acba7f6a6122b4bb/bindings/go/evmc/host.go#L233
Go byte array is converted to C pointer and returned in `evmc_result`. This seems wrong, because Go array is going to be garbage-collected, and then C has a dangling poiner.

Reply:
There the `C.evmc_make_result` makes a copy.

Reply:
Ah I see, missed that, thanks.

Reply:
In general, it looks you should pin the array `make([]C.evmc_tx_initcode, len(initcodes))` somewhere. Maybe add a private field `cInitcodes` to `TxContext`.